packages/core/src/actors.ts

```diff
@@ -244,25 +251,18 @@ export function fromObservable<T, TEvent extends EventObject>(
   const completeEventType = '$$xstate.complete';
 
   // TODO: add event types
-  const behavior: Behavior<
-    any,
-    T | undefined,
-    {
-      subscription: Subscription | undefined;
-      observable: Subscribable<T> | undefined;
-      canceled: boolean;
-      status: 'active' | 'done' | 'error';
-      data: T | undefined;
-    }
-  > = {
+  const behavior: Behavior<any, T | undefined, ObservableInternalState<T>> = {
     transition: (state, event, { self, id }) => {
       const _event = toSCXMLEvent(event);
+
       if (state.canceled) {
         return state;
       }
 
       switch (_event.name) {
         case nextEventType:
+          state.data = event.data.data;
+
           self._parent?.send(
```

Comment:
Those `self._parent.send` calls in this file should be wrapped with `actorCtx.defer` - at the same time we need to add tests for this. Those tests should do smth like:
```js
onDone: (ctx, ev) => {
  expect(child.getSnapshot()).toEqual(ev.data)
}
```